### ISSUE TITLE
Plex Update API Issue

### ISSUE THREAD

**Issue by dcm386** [taylorbourne/xteve_lazystream#6]
Based on what I see the plex web client doing, it looks like this request needs to be a POST. Also the X-Plex-Token=$plexToken needs to be included in the URL or else it results in a 401.

**Comment by taylor-verys** [comment]
I feel like I may have deleted some of the docs? Let me double check – I'll update you later on

**Comment by taylorbourne** [comment]
So this is working for me – every time I run the cronjob I see Plex updating it's guide data, no 401 reported.

```
### Plex
# Only necessary if xTeVe API is active
# Finding your Plex token: https://support.plex.tv/articles/204059436-finding-an-authentication-token-x-plex-token/
# Finding Plex ID: http://YOUR_IP_HERE:32400/?X-Plex-Token=YOUR_TOKEN_HERE , look for "tv.plex.providers.epg.xmltv:  " and enter the value blow.
plexIP="192.168.1.2"
plexPORT="32400"
plexToken=""
plexID=""
```

Are you sure you properly filled out this section of the cronjob file?

**Comment by dcm386** [comment]
Yes, it's definitely filled out. As it sits right now in cronjob.sh, curl is running in silent mode (-s), so it was failing with no messages on my machine. When I turned on verbose mode (-v instead of -s), and don't touch the curl command at all, I get a 401 response. When I changed "X-Plex-Client-Identifier" to "X-Plex-Token", I got a 404. When I forced the curl request to be a POST, I finally got a valid response from plex. So basically I changed the line in cronjob.sh to read:

`curl -v -X POST "http://$plexIP:$plexPORT/livetv/dvrs/$plexID/reloadGuide?X-Plex-Product=Plex%20Web&X-Plex-Version=4.8.4&X-Plex-Token=$plexToken&X-Plex-Platform=Firefox&X-Plex-Platform-Version=69.0&X-Plex-Sync-Version=2&X-Plex-Features=external-media&X-Plex-Model=bundled&X-Plex-Device=Linux&X-Plex-Device-Name=Firefox&X-Plex-Device-Screen-Resolution=1128x657%2C1128x752&X-Plex-Language=de" -H "User-Agent: Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:69.0) Gecko/20100101 Firefox/69.0" -H "Accept: text/plain, */*; q=0.01" -H "Accept-Language: de" --compressed -H "X-Requested-With: XMLHttpRequest" -H "Connection: keep-alive" -H "Referer: http://$plexIP:$plexPORT/web/index.html" --data ""`

I think some of that stuff can probably be removed, but I wanted to show what had worked for me with minimal changes to the cronjob.sh in the repo. I believe I'm using the latest public plex media server version (1.18.7.2457) if that matters at all (not plex pass).

**Comment by taylorbourne** [comment]
Hmm very nice, you comfortable opening up a pull request?

**Comment by dcm386** [comment]
Sure. I'll put one together after work today. I'll also experiment with removing some of the other stuff from that URL because I kind of doubt it needs things like screen resolution or browser type to work.

**Comment by dcm386** [comment]
PR https://github.com/taylorbourne/xteve_lazystream/pull/10 created

**Comment by taylorbourne** [comment]
Fixed with #10
